StratisMasternodeDashboard-master/Controllers/SidechainNodeController.cs

```diff
@@ -79,22 +79,37 @@ public async Task<IActionResult> ChangeLogLevelAsync(string rule, string level)
         }
 
         [Ajax]
-        [HttpPost] 
+        [HttpPost]
         [Route("vote")]
         public async Task<IActionResult> Vote(Vote vote)
         {
             if (string.IsNullOrEmpty(vote?.Hash))
                 return this.BadRequest("Hash is required");
-            
+
             ApiResponse response = await this.apiRequester.PostRequestAsync(this.defaultEndpointsSettings.SidechainNode, "/api/Voting/schedulevote-whitelisthash", new { hash = vote.Hash });
 
             if (response.IsSuccess) return this.Ok();
             if (response.Content?.errors != null)
             {
                 return this.BadRequest($"Failed to whitelist hash. Reason: {response.Content?.errors[0].message}");
             }
-            
+
             return this.BadRequest($"Failed to whitelist hash. Reason: {response.Content}");
+        }        
+
+        [HttpPost]
+        [Route("schedulekick")]
+        public async Task<IActionResult> ScheduleKick([FromBody] string pubKey)
+        {
+            if (string.IsNullOrEmpty(pubKey)) return this.BadRequest("Member key is required");
+            ApiResponse response = await this.apiRequester.PostRequestAsync(this.defaultEndpointsSettings.SidechainNode, "/api/Voting/schedulevote-kickmember", new { pubkey = pubKey });
+            if (response.IsSuccess) return this.Ok();
```

Comment:
It might better to rewrite the ifs like this:

```
            if (response.IsSuccess) 
                return this.Ok();

            if (response.Content?.errors != null)
                return this.BadRequest($"An error occurred trying to schedule a kick federation member vote: {response.Content?.errors[0].message}");

            return this.BadRequest($"An error occurred trying to schedule a kick federation member vote: {response.Content}");
```